src/uk/gov/defra/ffc/Helm.groovy

```diff
@@ -81,22 +84,26 @@ class Helm implements Serializable {
   }
 
   static def deployChart(ctx, environment, registry, chartName, tag) {
-    ctx.withKubeConfig([credentialsId: "kubeconfig-$environment"]) {
-      def deploymentName = "$chartName-$tag"
-      def extraCommands = getExtraCommands(tag)
-      def prCommands = getPrCommands(registry, chartName, tag, ctx.BUILD_NUMBER)
-
-      def helmValuesKeys = getHelmValuesKeys(ctx, "helm/$chartName/values.yaml")
-      def appConfigPrefix = environment + '/'
-      def defaultConfigValues = configItemsToSetString(getConfigValues(ctx, helmValuesKeys, appConfigPrefix))
-      def defaultConfigValuesChart = configItemsToSetString(getConfigValues(ctx, helmValuesKeys, appConfigPrefix, chartName))
-      def prConfigValues = configItemsToSetString(getConfigValues(ctx, helmValuesKeys, (appConfigPrefix + 'pr/')))
-      def prConfigValuesChart = configItemsToSetString(getConfigValues(ctx, helmValuesKeys, (appConfigPrefix + 'pr/'), chartName))
-
-      ctx.sh("kubectl get namespaces $deploymentName || kubectl create namespace $deploymentName")
-      ctx.echo('Running helm upgrade, console output suppressed')
-      ctx.sh("$suppressConsoleOutput helm upgrade $deploymentName --namespace=$deploymentName ./helm/$chartName $defaultConfigValues $defaultConfigValuesChart $prConfigValues $prConfigValuesChart $prCommands $extraCommands")
-      writeUrlIfIngress(ctx, deploymentName)
+    def repoName = Utils.getRepoName(ctx)
+    def commitSha = Utils.getCommitSha(ctx)
+    ctx.gitStatusWrapper(credentialsId: 'github-token', sha: commitSha, repo: repoName, gitHubContext: GitHubStatus.DeployChart.Context, description: GitHubStatus.DeployChart.Description) {
```

Comment:
Could you do the calls to `Utils.getRepoName` and `Utils.getCommitSha` in place rather than create variables for them each time which are only used once

Reply:
I didn't add them in place as I thought it made is clearer, however, I think it would be pretty clear doing it in place. I'll make the change.

As far as wrapping it into a function, the actions being done within the step would need to be wrapped up in a function to be executed within the step. I'm not sure it would be worth the effort of adding the complexity of that for reduced duplication of the call.